packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -185,11 +193,9 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         this.sendTelemetryEventCore(perfEvent, error);
     }
 
-    protected prepareEvent(event: ITelemetryBaseEvent): ITelemetryBaseEvent {
+    protected prepareEvent(event: ITelemetryBaseEventExt): ITelemetryBaseEventExt {
         const includeErrorProps = event.category === "error" || event.error !== undefined;
-        const newEvent: ITelemetryBaseEvent = {
-            ...event,
-        };
+        const newEvent: ITelemetryBaseEventExt = convertToBaseEvent(event) as ITelemetryBaseEventExt;
```

Comment:
Shouldn't need to do this, or really change this function at all.  It's dealing in the base event type and should stay that way (by this time extended events have already been converted)

Reply:
I reverted back to not calling `convertToBaseEvent(event)`, but would this be okay for new logger classes that extend `TelemetryLogger` and implement it's own send shown in the example below?

```
class TestTelemetryLogger extends TelemetryLogger {
    public events: ITelemetryBaseEventExt[] = [];
    public send(event: ITelemetryBaseEventExt): void {
        this.events.push(this.prepareEvent(event));
    }
}
```

If the extended event is passed through directly into the `prepareEvent()` function (as shown above), it will not be converted yet and lead to an error.

Reply:
Not sure I follow so lmk if I'm missing it.  But no one should ever override `send` and change its type - we just add the specialized `sendFoo` functions that take the extended stuff.  But `send` itself needs to stay basic.